Topics/Software_Engineering/Agile.md

```diff
@@ -0,0 +1,98 @@
+
+2. **Working software over comprehensive documentation**:
+    - Agile teams focus on delivering working software that provides value to the customer. While documentation is important, it is kept to a minimum and is only created when necessary. The main goal is to deliver a functioning product, not to produce extensive documentation.
+    - **Example**: In a traditional waterfall model, extensive documentation is often produced before any code is written. In an Agile project, the emphasis is on delivering working software at the earliest opportunity. For instance, instead of spending weeks creating detailed system documentation, the Agile team might prioritize delivering a minimum viable product (MVP) quickly. This allows stakeholders to interact with the actual software sooner, providing valuable insights and facilitating iterative improvements.
+
+3. **Customer collaboration over contract negotiation**:
+    - Agile teams prioritize regular and direct communication with the customer to understand their needs and expectations. This collaboration helps in delivering software that aligns with the customer's requirements. It's not just about delivering what's in the contract, but working together to ensure the product meets the customer's needs.
+    - **Example**: Imagine a software development team is working on an e-commerce platform. In a traditional, contract-centric approach, the team might strictly adhere to the initially agreed-upon features and functionalities. However, in an Agile environment, the team regularly collaborates with the customer to gather feedback. If the customer expresses a need for additional features like a personalized recommendation system, the Agile team can quickly adjust the development roadmap to incorporate these changes, ensuring the final product better meets the customer's vision.
+
+4. **Responding to change over following a plan**:
+    - Agile teams embrace change and are flexible in adapting to new requirements or priorities. They value feedback and continuously improve the software based on customer input. Instead of sticking rigidly to a plan, Agile teams are ready to change direction if it benefits the project.
+    - **Example**: Consider a mobile app development project where the initial plan outlines a specific set of features to be delivered in each development phase. In an Agile setting, if user feedback or market trends indicate a need for a fundamental change, such as a shift in the user interface design, the Agile team can swiftly adapt. Instead of rigidly sticking to the original plan, Agile allows for continuous adjustments, ensuring the final product is not only functional but also aligned with the latest user preferences.
+
+## Agile Manifesto Principles
```

Comment:
While I feel like you explained the 12 principles well, I think the original text was already clear enough and didn't need to be explained further. Perhaps just list the principles and give a short summary of all of them at the end?

Reply:
I see that you have other comments on the examples, but overall, do you think it might be better to remove the examples? You point is valid, but I think giving a short summary of each principle as I list them might be easier to read and digest, rather than one short summary of all of them at the end.